scaleway/data_source_instance_ip.go

```diff
@@ -0,0 +1,62 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func dataSourceScalewayInstanceIP() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceScalewayInstanceIP().Schema)
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "address")
+	dsSchema["id"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "The ID of the IP address",
+		ValidateFunc: validationUUIDWithLocality(),
+	}
+	dsSchema["address"].ConflictsWith = []string{"id"}
```

Comment:
Can we treat this schema as here https://github.com/scaleway/terraform-provider-scaleway/blob/aa12d5d0b5f90d3ee5d480c53a387004c43ec7b9/scaleway/data_source_lb_ip.go#L17

Reply:
You mean with the `ip_` prefix ?